packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -339,6 +339,19 @@ class FlutterPlugin implements Plugin<Project> {
                 implementation("androidx.multidex:multidex:2.0.1")
             }
             }
+
+        // Validate that the provided Gradle, Java, AGP, and KGP versions are all within our
+        // supported range.
+        if (!project.hasProperty("skipDependencyChecks") || !project.getProperty("skipDependencyChecks")) {
+            try {
+                project.apply from: Paths.get(flutterRoot.absolutePath, "packages", "flutter_tools",
```

Comment:
Consider making the path a final or constant in the file.

Reply:
Changed